api-mobile/src/paths/activity.ts

```diff
@@ -0,0 +1,234 @@
+
+import { ManagedUpload } from 'aws-sdk/clients/s3';
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { SQLStatement } from 'sql-template-strings';
+import { WRITE_ROLES } from '../constants/misc';
+import { getDBConnection } from '../database/db';
+import { ActivityPostBody, IMediaItem, MediaBase64 } from '../models/activity';
+import { postActivitySQL } from '../queries/activity-queries';
+import { uploadFileToS3 } from '../utils/file-utils';
+import { getLogger } from '../utils/logger';
+
+const defaultLog = getLogger('activity-controller');
+
+export const POST: Operation = [uploadMedia(), createActivity()];
+
+POST.apiDoc = {
+  description: 'Create a new activity.',
+  tags: ['activity'],
+  security: [
+    {
+      Bearer: WRITE_ROLES
+    }
+  ],
+  requestBody: {
+    description: 'Activity post response object.',
+    content: {
+      'application/json': {
+        schema: {
+          required: [
+            'activityType',
+            'activityTypeData',
+            'activitySubType',
+            'activitySubTypeData',
+            'date',
+            'locationAndGeometry'
+          ],
+          properties: {
+            activityType: {
+              type: 'string'
+            },
+            activityTypeData: {
+              type: 'object'
+            },
+            activitySubType: {
+              type: 'string'
+            },
+            activitySubTypeData: {
+              type: 'object'
+            },
+            date: {
+              type: 'string',
+              description: 'Date in YYYY-MM-DD format'
+            },
+            locationAndGeometry: {
+              type: 'object',
+              properties: {
+                anchorPointY: {
+                  type: 'number'
+                },
+                anchorPointX: {
+                  type: 'number'
+                },
+                area: {
+                  type: 'number'
+                },
+                geometry: {
+                  type: 'object',
+                  description: 'A geoJSON object'
+                },
+                jurisdiction: {
+                  type: 'string'
+                },
+                agency: {
+                  type: 'string'
+                },
+                observer1FirstName: {
+                  type: 'string'
+                },
+                observer1LastName: {
+                  type: 'string'
+                },
+                locationComment: {
+                  type: 'string'
+                },
+                generalComment: {
+                  type: 'string'
+                },
+                photoTaken: {
+                  type: 'boolean'
+                }
+              }
+            },
+            media: {
+              description: 'The keys for the uploaded files',
+              type: 'array',
+              items: {
+                type: 'object',
+                required: ['fileName', 'encodedFile'],
```

Comment:
I'm not seeing where the filename lives once the data hits the table, is it only stored in s3?

Reply:
Good point, I think I missed including that.

We can chat tomorrow (Friday) about how we want to store the media information.  Maybe another json blob column, that can hold any number of media files and their meta?  Can chat tomorrow.

Reply:
Oh, it is in the raw payload we store as well. 

The media array that gets sent in the request `[ { fileName: string, encodedFile: string } ]` gets saved, just with the `encodedFile` property stripped out so it doesn't explode the json blob.

Seen here: https://github.com/bcgov/lucy-web/pull/882/files#diff-99a086dc30892f0c40593bfccbe28751R75-R80